From 4c351abfc3acc81c64a0317b858da7c13ec7635e Mon Sep 17 00:00:00 2001
From: Yu Ming Zhu <yzhu@redhat.com>
Date: Wed, 15 Jul 2020 13:42:50 +0000
Subject: [PATCH 06/20] do not break things when mergerepo_c version is not
 detectable

---
 builder/kojid | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/builder/kojid b/builder/kojid
index a1c39ad66d0d..1a120c80542b 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -5555,7 +5555,8 @@ class CreaterepoTask(BaseTaskHandler):
             blocklist = self.repodir + '/blocklist'
             cmd.extend(['-b', blocklist])
         cmd.extend(['-a', arch, '-o', self.outdir])
-        if cmd[0].endswith('mergerepo_c') and mergerepo_c_version >= (0, 15, 11):
+        if cmd[0].endswith('mergerepo_c') and mergerepo_c_version \
+           and mergerepo_c_version >= (0, 15, 11):
             cmd.append('--arch-expand')
         if os.path.isfile(groupdata):
             cmd.extend(['-g', groupdata])
-- 
2.52.0

