From c164973315f5788be9aa265624094b5c079b2e79 Mon Sep 17 00:00:00 2001
From: Yu Ming Zhu <yzhu@redhat.com>
Date: Wed, 15 Jul 2020 17:54:21 +0000
Subject: [PATCH 08/20] depress Exception when checking mergerepo_c version

---
 builder/kojid | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/builder/kojid b/builder/kojid
index f4c5986fad81..9585e347ab51 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -5482,10 +5482,14 @@ class CreaterepoTask(BaseTaskHandler):
 
     def _get_mergerepo_c_version(self):
         cmd = ['/usr/bin/mergerepo_c', '--version']
-        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-        out, _ = proc.communicate()
-        status = proc.wait()
-        if status != 0:
+        try:
+            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+            out, _ = proc.communicate()
+            status = proc.wait()
+            if status != 0:
+                self.logger.warning("Unable to detect mergerepo_c version")
+                return None
+        except Exception:
             self.logger.warning("Unable to detect mergerepo_c version")
             return None
         out = out.decode().strip()
-- 
2.52.0

