From a212aff665dd32b38ecca54c5f07bd375b066a3b Mon Sep 17 00:00:00 2001
From: Mike McLean <mikem@redhat.com>
Date: Tue, 8 Nov 2022 13:08:15 -0500
Subject: [PATCH 14/20] Emulate old list data

---
 koji/rpmdiff.py | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/koji/rpmdiff.py b/koji/rpmdiff.py
index 96b7b09f9e8d..e8b386b920de 100644
--- a/koji/rpmdiff.py
+++ b/koji/rpmdiff.py
@@ -231,12 +231,30 @@ class Rpmdiff:
             return self.__fileIteratorToDict(hdr.fiFromHeader())
         result = {}
         for file in rpm.files(hdr):
-            result[file.name] = file
+            # mimic old fi order because comparison and kojihash relies on it
+            # (FN, FSize, FMode, FMtime, FFlags, FRdev, FInode, FNlink, FState, VFlags, FUser, FGroup, Digest)
+            filedata = [
+                    # name omitted
+                    file.size,
+                    file.mode,
+                    file.mtime,
+                    file.fflags,
+                    file.rdev,
+                    file.inode,
+                    file.nlink,
+                    file.state,
+                    file.vflags,
+                    file.user,
+                    file.group,
+                    file.digest,
+                    ]
+            result[file.name] = filedata
         return result
 
     def __fileIteratorToDict(self, fi):
         result = {}
         for filedata in fi:
+            # index by name
             result[filedata[0]] = list(filedata[1:])
         return result
 
-- 
2.52.0

