From 3c5ddb8dab9a5736f03b6d60593adc9170304af3 Mon Sep 17 00:00:00 2001
From: Mike McLean <mikem@redhat.com>
Date: May 03 2019 14:57:43 +0000
Subject: [PATCH 1/2] handle bare merge mode


Fixes https://pagure.io/koji/issue/1404

---

diff --git a/builder/kojid b/builder/kojid
index 5110d04..e176266 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -5186,6 +5186,10 @@ class CreaterepoTask(BaseTaskHandler):
             cmd = ['/usr/libexec/kojid/mergerepos',
                    '--mode', 'simple',
                    '--tempdir', self.workdir]
+        elif merge_mode == 'bare':
+            # "bare" merge mode for repos with modular metadata
+            # forces use of mergerepo_c
+            cmd = ['/usr/bin/mergerepo_c', '--pkgorigins']
         elif self.options.use_createrepo_c:
             cmd = ['/usr/bin/mergerepo_c', '--koji']
         else:
diff --git a/koji/__init__.py b/koji/__init__.py
index 6e667ce..9c7101c 100644
--- a/koji/__init__.py
+++ b/koji/__init__.py
@@ -236,7 +236,7 @@ REPO_EXPIRED = REPO_STATES['EXPIRED']
 REPO_DELETED = REPO_STATES['DELETED']
 REPO_PROBLEM = REPO_STATES['PROBLEM']
 
-REPO_MERGE_MODES = set(['koji', 'simple'])
+REPO_MERGE_MODES = set(['koji', 'simple', 'dumb'])
 
 # buildroot states
 BR_STATES = Enum((

From 0da20019456c386aa44589c3f0529926fb426541 Mon Sep 17 00:00:00 2001
From: Mike McLean <mikem@redhat.com>
Date: May 03 2019 16:54:54 +0000
Subject: [PATCH 2/2] fix mode name


---

diff --git a/koji/__init__.py b/koji/__init__.py
index 9c7101c..0b9b5bc 100644
--- a/koji/__init__.py
+++ b/koji/__init__.py
@@ -236,7 +236,7 @@ REPO_EXPIRED = REPO_STATES['EXPIRED']
 REPO_DELETED = REPO_STATES['DELETED']
 REPO_PROBLEM = REPO_STATES['PROBLEM']
 
-REPO_MERGE_MODES = set(['koji', 'simple', 'dumb'])
+REPO_MERGE_MODES = set(['koji', 'simple', 'bare'])
 
 # buildroot states
 BR_STATES = Enum((

