From 403f0f8a44c8ee0fd44326ae4d11bafe79e1809a Mon Sep 17 00:00:00 2001
From: Patrick Uiterwijk <patrick@puiterwijk.org>
Date: May 29 2019 15:31:52 +0000
Subject: Allow builder to attempt krb if gssapi is available


Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

---

diff --git a/builder/kojid b/builder/kojid
index be1cb26..91d4d69 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -69,6 +69,11 @@ except ImportError:  # pragma: no cover
     krbV = None
 
 try:
+    import requests_kerberos
+except ImportError:  # pragma: no cover
+    requests_kerberos = None
+
+try:
     import librepo
     import io
 except ImportError:
@@ -6137,7 +6142,7 @@ if __name__ == "__main__":
             quit("Error: Unable to log in. Bad credentials?")
         except six.moves.xmlrpc_client.ProtocolError:
             quit("Error: Unable to connect to server %s" % (options.server))
-    elif krbV:
+    elif krbV or requests_kerberos:
         krb_principal = options.krb_principal
         if krb_principal is None:
             krb_principal = options.host_principal_format % socket.getfqdn()

