diff -Nur koji-1.18.0.orig/cli/koji.conf koji-1.18.0/cli/koji.conf
--- koji-1.18.0.orig/cli/koji.conf	2019-08-09 11:07:07.000000000 -0700
+++ koji-1.18.0/cli/koji.conf	2019-08-16 08:14:51.834690613 -0700
@@ -4,17 +4,22 @@
 
 ;url of XMLRPC server
 ;server = http://hub.example.com/kojihub
+server = https://koji.fedoraproject.org/kojihub
 
 ;url of web interface
 ;weburl = http://www.example.com/koji
+weburl = https://koji.fedoraproject.org/koji
 
 ;url of package download site
 ;pkgurl = http://www.example.com/packages
+pkgurl = https://kojipkgs.fedoraproject.org/
 
 ;path to the koji top directory
 ;topdir = /mnt/koji
 
 ;configuration for Kerberos authentication
+authtype = kerberos
+krb_rdns = false
 
 ;the service name of the principal being used by the hub
 ;krbservice = host
@@ -28,7 +33,6 @@
 ;enable to lookup dns canonical hostname for krb auth
 ;krb_canon_host = no
 
-
 ;configuration for SSL authentication
 
 ;client certificate
@@ -45,9 +49,14 @@
 ;[not_implemented_yet]
 ;enabled plugins for CLI, runroot and save_failed_tree are available
 ;plugins =
+; runroot plugin is enabled by default in fedora
+plugins = runroot
 
 ;timeout of XMLRPC requests by seconds, default: 60 * 60 * 12 = 43200
 ;timeout = 43200
 
 ;timeout of GSSAPI/SSL authentication by seconds, default: 60
 ;auth_timeout = 60
+
+; use the fast upload feature of koji by default
+use_fast_upload = yes
