diff -rup koji-1.15.0.orig/cli/koji.conf koji-1.15.0/cli/koji.conf
--- koji-1.15.0.orig/cli/koji.conf	2018-01-27 10:44:39.111029362 +0100
+++ koji-1.15.0/cli/koji.conf	2018-01-27 10:46:33.713817186 +0100
@@ -3,18 +3,23 @@
 ;configuration for koji cli tool
 
 ;url of XMLRPC server
-;server = http://hub.example.com/kojihub
+server = https://koji.fedoraproject.org/kojihub
 
 ;url of web interface
-;weburl = http://www.example.com/koji
+weburl = https://koji.fedoraproject.org/koji
 
 ;url of package download site
-;pkgurl = http://www.example.com/packages
+topurl = https://kojipkgs.fedoraproject.org/
 
 ;path to the koji top directory
 ;topdir = /mnt/koji
 
+; use the fast upload feature of koji by default
+use_fast_upload = yes
+
 ;configuration for Kerberos authentication
+authtype = kerberos
+krb_rdns = false
 
 ;the service name of the principal being used by the hub
 ;krbservice = host
@@ -29,13 +34,6 @@
 ;krb_canon_host = no
 
 
-;configuration for SSL authentication
-
-;client certificate
-;cert = ~/.koji/client.crt
-
-;certificate of the CA that issued the HTTP server certificate
-;serverca = ~/.koji/serverca.crt
-
-;enabled plugins for CLI, runroot and save_failed_tree are available
-;plugins =
+;enabled plugins for CLI, runroot is enabled by deafult for fedora
+;save_failed_tree is available
+plugins = runroot
